cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/TableUpdater.java

```diff
@@ -70,9 +68,11 @@ public TableUpdater(String rowType, final Configuration conf) {
    * @param regionName is the key for the Map
    * @param time is the value for the Map
    */
-  public synchronized void updateTime(String regionName, long time) throws IOException {
-    if (!cachedUpdates.containsKey(regionName) || time > cachedUpdates.get(regionName)) {
-      cachedUpdates.put(regionName, time);
+  public void updateTime(String regionName, long time) throws IOException {
+    synchronized (cachedUpdates) {
```

Comment:
In a single co-processor consecutive calls to postWAL/postReplicate are handled in separate threads so the map update can happen in any order by these threads. The updateTime method had to be synchronized because it need to check the entry and update only when time > cachedUpdates.get(regionName), The updateTime method was synchronized in the older changes already so there should be no performance impact there. The only performance impact now is the extra clone which only happens once every 20secs.